src/utils/parser/sources.js

```diff
@@ -0,0 +1,24 @@
+// @flow
+
+import type { Source, SourceId } from "debugger-html";
+
+let cachedSources = new Map();
+
+export function hasSource(sourceId: SourceId): boolean {
+  return cachedSources.has(sourceId);
+}
+
+export function setSource(source: Source) {
+  cachedSources.set(source.id, source);
+}
+
+export function getSource(sourceId: SourceId): Source {
+  if (!cachedSources.has(sourceId)) {
+    throw new Error(`${sourceId} was not provided.`);
+  }
+  return ((cachedSources.get(sourceId): any): Source);
+}
+
+export function clearSources() {
+  cachedSources = new Map();
+}
```

Comment:
almost all of our APIs now expect to get a `source`. 

I agree that this is a performance issue, but I think we should make the fix in a separate PR, that touches all of the spots.

https://github.com/devtools-html/debugger.html/blob/master/src/utils/parser/steps.js#L6-L8
https://github.com/devtools-html/debugger.html/blob/master/src/utils/parser/getSymbols.js#L335-L336

#3953